File: sixth-form/notes/stem/cs/as-and-a-level-ocr/onenote-structure/unit2/12-algorithms/code-examples/search/linear/linear.py
def linear_search(input_list: list, target: any) -> any:
    for i in range(len(input_list)):
        if input_list[i] == target:
            return i  # return the index of the target


def main():
    nums = [2, 4, 32, 34, 5, 2, 12, 3123, 13, 243, 234]

    print(linear_search(nums, 13))


if __name__ == "__main__":
    main()

File: sixth-form/notes/stem/cs/as-and-a-level-ocr/onenote-structure/unit2/12-algorithms/code-examples/search/linear/linear.c
#include <stdio.h>

// pass the array by reference to the function
int linear_search(int *arr, int target) {

  // dereference the pointer when working out the size to look at the array at
  // the corresponding location in memory
  int arr_len = sizeof(&arr) / sizeof(&arr[0]);

  for (int i = 0; i < arr_len; i++) {
    if (arr[i] == target) {
      return i;
    }
  }
  return -1;
}

int main() {
  int target = 5;

  int nums[] = {5, 4, 3, 2, 1};

  /* int result = linear_search(nums, target); */

  printf("index of %d is %d", target, linear_search(nums, target));
}
